pkg/devserver/devserver.go

```diff
@@ -93,33 +93,35 @@ func start(ctx context.Context, opts StartOpts) error {
 
 	stepLimitOverrides := make(map[string]int)
 
-	rc, err := createInmemoryRedis(ctx, opts.Tick)
+	shardedRc, err := createInmemoryRedis(ctx, opts.Tick)
 	if err != nil {
 		return err
 	}
 
+	unshardedRc, err := createInmemoryRedis(ctx, opts.Tick)
+	if err != nil {
+		return err
+	}
+
+	unshardedClient := redis_state.NewUnshardedClient(unshardedRc, redis_state.StateDefaultKey, redis_state.QueueDefaultKey)
+	shardedClient := redis_state.NewShardedClient(unshardedClient, shardedRc, redis_state.StateDefaultKey)
+
 	var sm state.Manager
 	t := runner.NewTracker()
 	sm, err = redis_state.New(
 		ctx,
-		redis_state.WithRedisClient(rc),
-		redis_state.WithKeyGenerator(redis_state.DefaultKeyFunc{
-			Prefix: "{state}",
-		}),
+		redis_state.WithShardedClient(shardedClient),
+		redis_state.WithUnshardedClient(unshardedClient),
 	)
 	if err != nil {
 		return err
 	}
 	smv2 := redis_state.MustRunServiceV2(sm)
 
-	queueKG := &redis_state.DefaultQueueKeyGenerator{
-		Prefix: "{queue}",
```

Comment:
Instead of defining a global fallback and specific values for key prefixes, we reuse constants and require them at the time of client creation.